File: hackerrank/Data Structures/Linked Lists/02 — Insert a Node at the Tail of a Linked List/solution.py
# Source : https://www.hackerrank.com/challenges/insert-a-node-at-the-tail-of-a-linked-list/problem
# Author : Frank Faustino
# Date   : 2019-01-19

"""
 Insert Node at the end of a linked list 
 head pointer input could be None as well for empty list
 Node is defined as
 
 class Node(object):
 
   def __init__(self, data=None, next_node=None):
       self.data = data
       self.next = next_node
 
 return back the head of the linked list in the below method
"""

def Insert(head, data):
    if head is None:
        head = Node(data)
    else:
        current = head
        while current.next:
            current = current.next
        current.next = Node(data)
    return head

File: hackerrank/Data Structures/Linked Lists/02 — Insert a Node at the Tail of a Linked List/solution.cs
// Source : https://www.hackerrank.com/challenges/insert-a-node-at-the-tail-of-a-linked-list/problem
// Author : Frank Faustino
// Date   : 2018-01-19
// Lang   : C#

/*
    Insert Node at the end of a linked list
    head pointer input could be NULL as well for empty list
    Node is defined as
    class Node {
        int Data;
        Node Next;
    }
*/
// This is a "method-only" submission.
// You only need to complete this method.

public static Node Insert(Node head, int x) {
    if (head == null) {
        head = new Node();
        head.Data = x;
    }
    else {
        Node node = head;
        while (node.Next != null)
            node = node.Next;
        node.Next = new Node();
        node.Next.Data = x;
    }
    return head;
}
